library/patient_self_checkin/step5.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+use OpenEMR\Core\Header;
+
+$ignoreAuth = true;
+require_once "../../interface/globals.php";
+if (!$GLOBALS['self_checkin_enable'] == 1) {
+    die("This feature has not been enabled");
+}
+?>
+
+<link rel="stylesheet" type="text/css" href="/interface/themes/selfCheckIn.css">
+
+<!DOCTYPE html>
+<html>
+<head>
+
+    <title><?php echo xlt("Check In"); ?></title>
+<?php
+
+Header::setupHeader();
+
+?>
+</head>
+<body class="step5">
+
+<p>
+    <img style="text-align: center"
+         width="500px" src='<?php echo $GLOBALS['images_static_relative']; ?>/logo-full-con.png'/>
+</p>
+
+<?php
+
+$month = htmlspecialchars($_GET['month']);
+$birthday = htmlspecialchars($_GET['date']);
+$surname = htmlspecialchars($_GET['surname']);
+
+// date_default_timezone_set();
+
+$timeRN = date("Y-m-d H:i:s");
+$date = date("y-m-d");
+echo "<h1> Time: " . $timeRN . "</h1><br>";
+
+require 'mySQL_connector.php';
+
+// Get the patient IDs of everyone with an appointment today. Enter them into an array which we can call upon later.
+
+$todaysPIDsQuery = "SELECT pc_pid from `openemr_postcalendar_events` WHERE `pc_eventDate` = '" . $date ."'";
+
+$todaysPIDs_result = mysqli_query($dbc, $todaysPIDsQuery); // This line executes the MySQL query that you typed above
+
+$PIDArray = array(); // make a new array to hold all your data
+
+$index = 0;
+while ($row = mysqli_fetch_array($todaysPIDs_result)) { // loop to store the data in an associative array.
+     $PIDArray[$index] = $row;
+     $index++;
+}
```

Comment:
The majority of the business logic on this script can be covered by several already existent functions. Will go through that later. But here's an example code block above that uses OpenEMR sql wrapper and binding for sql injection:
``````
$todaysPIDsQuery = "SELECT pc_pid from `openemr_postcalendar_events` WHERE `pc_eventDate` = ?";
$todaysPIDs_result = sqlStatement($todaysPIDsQuery, array($date));

$PIDArray = array(); // make a new array to hold all your data

$index = 0;
while ($row = sqlFetchArray($todaysPIDs_result)) { // loop to store the data in an associative array.
     $PIDArray[$index] = $row;
     $index++;
}
``````